### PR TITLE
#162 using external file for aws variables

### PR DESCRIPTION
**Pull request by ynixon**
#162

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -29,16 +74,17 @@ Vagrant.configure("2") do |config|
...
-			override.ssh.username = "ubuntu"
-			override.ssh.private_key_path = "~/.ssh/osx_rsa.pem"
+			aws.keypair_name = AWS['stage_keypair_name']
+			aws.ami = AWS['stage_ami']
+			aws.instance_type = AWS['stage_instance_type']
+			aws.region = AWS['stage_region']
+			aws.subnet_id = AWS['stage_subnet_id']
+			aws.security_groups = AWS['stage_security_groups']
+			aws.associate_public_ip = AWS['stage_associate_public_ip']
+			aws.access_key_id = AWS['stage_access_key_id']
```
Comment (lioramilbaum) [inline Vagrantfile]: the access_key_id and the secret_access_key is not needed.  will explain in our coming AWS workshop.

File: Vagrantfile [hunk 2]
```diff
@@ -29,16 +74,17 @@ Vagrant.configure("2") do |config|
...
+			aws.secret_access_key = AWS['stage_secret_access_key']
```
Comment (lioramilbaum) [inline Vagrantfile]: prev comment

File: Vagrantfile [hunk 3]
```diff
@@ -50,17 +96,18 @@ Vagrant.configure("2") do |config|
...
-			override.ssh.private_key_path = "~/.ssh/osx_rsa.pem"
+			aws.keypair_name = AWS['prod_keypair_name']
+			aws.ami = AWS['prod_ami']
+			aws.instance_type = AWS['prod_instance_type']
+			aws.elastic_ip = AWS['prod_elastic_ip']
+			aws.region = AWS['prod_region']
+			aws.subnet_id = AWS['prod_subnet_id']
+			aws.security_groups = AWS['prod_security_groups']
+			aws.associate_public_ip = AWS['prod_associate_public_ip']
+			aws.access_key_id = AWS['prod_access_key_id']
```
Comment (lioramilbaum) [inline Vagrantfile]: prev comment

File: Vagrantfile [hunk 4]
```diff
@@ -50,17 +96,18 @@ Vagrant.configure("2") do |config|
...
+			aws.secret_access_key = AWS['prod_secret_access_key']
```
Comment (lioramilbaum) [inline Vagrantfile]: prev comment

File: aws.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+---
+stage_keypair_name: "osx_rsa"
+stage_ami: "ami-d2414e38"
+stage_instance_type: "t2.micro"
+stage_region: "eu-west-1"
+stage_subnet_id: "subnet-2c67fe64"
+stage_security_groups: ["sg-7b78fe07"]
+stage_associate_public_ip: true
+stage_access_key_id: ""
```
Comment (lioramilbaum) [inline aws.yml]: not needed

File: aws.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+stage_secret_access_key: ""
```
Comment (lioramilbaum) [inline aws.yml]: same

File: aws.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+prod_keypair_name: "osx_rsa"
+prod_ami: "ami-d2414e38"
+prod_instance_type: "t2.micro"
+prod_elastic_ip: "52.209.230.146"
+prod_region: "eu-west-1"
+prod_subnet_id: "subnet-2c67fe64"
+prod_security_groups: ["sg-7b78fe07"]
+prod_associate_public_ip: true
+prod_access_key_id: ""
```
Comment (lioramilbaum) [inline aws.yml]: same

File: aws.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+prod_secret_access_key: ""
```
Comment (lioramilbaum) [inline aws.yml]: same

File: lib/vagrant.rb [hunk 9]
```diff
@@ -0,0 +1,17 @@
+module OS
```
Comment (lioramilbaum) [inline lib/vagrant.rb]: there is a ruby gem names os which doing the same. it is reinventing the weal?

Comment (ynixon) [reply lib/vagrant.rb]: I have just looked for this, 
If I will try to use a missing gem, I would need to automatically install it.
(I assume I will need another extra gem for this)
When we will have an external wrapper, we will be able to install it externally.
I couldn't find other a proper way.

File: Vagrantfile (hunk 1 above)
Comment (ynixon) [reply Vagrantfile]: found the reason, fixed

File: lib/vagrant.rb (hunk 9 above)
Comment (lioramilbaum) [reply lib/vagrant.rb]: then use a different approach:
https://stackoverflow.com/questions/26811089/vagrant-how-to-have-host-platform-specific-provisioning-steps

File: aws.yml (hunk 5 above)
Comment (ynixon) [reply aws.yml]: security_groups and subnet_id are different between users, no need to overwrite them each pull

File: lib/vagrant.rb (hunk 9 above)
Comment (ynixon) [reply lib/vagrant.rb]: LOL, this is exactly where I took my solution from 
but didn't looked down :(
fixed

Comment (ynixon) [reply lib/vagrant.rb]: done

File: Vagrantfile [hunk 10]
```diff
@@ -1,8 +1,50 @@
...
 # vi: set ft=ruby :
 
+require 'yaml'
+AWS = YAML.load_file 'aws.yml'
+
+puts ""
+
+if Vagrant::Util::Platform.windows?
+    puts "Vagrant was launched from Windows."
+elsif Vagrant::Util::Platform.mac?
```
Comment (lioramilbaum) [inline Vagrantfile]: doesn't work on my mac. maybe switch to .cygwin?

Comment (lioramilbaum) [reply Vagrantfile]: you probably want to switch the 'mac' with 'darwin'

Comment (ynixon) [reply Vagrantfile]: can you try:
puts Vagrant::Util::Platform

Comment (lioramilbaum) [reply Vagrantfile]: I have used 'puts Vagrant::Util::Platform.platform'. The output is: darwin13.4.0

Comment (ynixon) [reply Vagrantfile]: removed the "Vagrant was launched from..."

File: Vagrantfile [hunk 11]
```diff
@@ -1,8 +1,38 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+require 'yaml'
+AWS = YAML.load_file 'aws.yml'
+
+puts ""
```
Comment (lioramilbaum) [inline Vagrantfile]: Is there a real need for this extra blank line?

Comment (ynixon) [reply Vagrantfile]: removed

**Pull request merged**